pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -194,12 +197,20 @@ func (i *instantSplitter) mapCall(expr *parser.Call) (mapped parser.Expr, finish
 		return i.mapCallAvgOverTime(expr)
 	case countOverTime:
 		return i.mapCallVectorAggregation(expr, parser.SUM)
+	case increase:
+		return i.mapCallVectorAggregation(expr, parser.SUM)
+	case irate:
+		return i.mapCallRate(expr)
 	case maxOverTime:
 		return i.mapCallVectorAggregation(expr, parser.MAX)
 	case minOverTime:
 		return i.mapCallVectorAggregation(expr, parser.MIN)
+	case presentOverTime:
+		return i.mapCallVectorAggregation(expr, parser.MAX)
```

Comment:
Using the aggregator `MAX` here is a hack, `MIN` could also be used. Since `present_over_time` returns the value 1 for any series in the specified interval, this just enforces that all 1 values are returned. Let me know if you see any problems with this 🪄